### ISSUE TITLE
node_modules is required for production build

### ISSUE THREAD

**Issue by omerts** [ctrlplusb/react-universally#93]
Hello,

Thanks for the great boilerplate.
We would like to be able to build our project into the `build` folder, and allow that folder to be deployed alone. Currently the bundle file has requires for packages from the node_modules folder, and doesn't include the packages inside the bundle itself. This makes node_modules folder required, as part of deploying the app. 
Node modules folder contains a lot of redundant files, causing our deployment package to be hundreds of time larger than it should be. (Node modules folder is a few hundred Mbs big).

There is also an issue that the client assets file is being looked up using `\build\client\assets.json`, which means you must be running from outside the build folder. 

Imho, a build folder should be more of a dist folder, where the whole app can run from it, without needing any other assets.

**Comment by ctrlplusb** [comment]
Hi @omerts 

Okay, yeah, I see what you are going for here.  I can have a play around with the concept of creating a full encapsulated ___dist___.

I do hold concerns that it could be quite chunky in size, however, I made some recent changes to the webpack integration which may now alleviate this.

I'll ping you back with my findings and we can discuss where best this feature should be placed.

**Comment by ctrlplusb** [comment]
Hi @omerts 

I have made some progress on this.  I was right in assuming some previous changes I made would make this process much easier.

There is one outstanding issue though before I can release this new feature.  It has to do with a 3rd party library unfortunately.  

I'll keep this thread updated.

**Comment by ctrlplusb** [comment]
Hey @omerts 

An update on this.  Unfortunately even after solving an issue with a package that was causing the bundle to break there appears to be multiple issues with some of the other node modules, one of them being `express`, which do some dynamic requires (e.g. `require(fooVar)`).  Unfortunately the webpack bundling process doesn't support this so we are blocked on this issue for now... :'(

There are some related threads in the webpack issues.  If I come across them again I will link them here.

**Comment by omerts** [comment]
@ctrlplusb Thanks for the update. We too tried checking it out, and run into problems with the helmet package, doing some dynamic requires, that have no webpack context whatsoever. We tried using the webpack's ContextReplacementPlugin, but with no success. I also opened a question on stackoverflow, but it still has no answers: [Stackoverflow](http://stackoverflow.com/questions/39353001/webpack-contextreplacementplugin-get-packages-from-node-modules)
